src/graph.rs

```diff
@@ -112,6 +121,65 @@ where
     }
 }
 
+/// Exclusion set `S` of nodes that are removed from `G`. Encapsulated in this
+/// interface to evaluate optimizations to its implementation (e.g., set vs vec).
+#[derive(Debug, Clone, PartialEq)]
+pub struct ExclusionSet {
```

Comment:
This is a struct, not a trait.
Reason I'm saying that is that we could also try the `BitSet` collection to check performance.